src/components/authentication-error-dialog/index.tsx

```diff
@@ -0,0 +1,30 @@
+import useAuthentication from 'hooks/use-authentication'
+import Button from 'components/button'
+import Dialog from 'components/dialog'
+import Text from 'components/text'
+import s from './authentication-error-dialog.module.css'
+import { IconSignIn16 } from '@hashicorp/flight-icons/svg-react/sign-in-16'
+
+const AuthenticationErrorDialog = () => {
```

Comment:
[question/suggestion] It might be a good idea to add a segment event here so we can get an idea of how often the modal is triggering. Ideally I think the modal is a last resort that we want to see fire rarely, if ever.

I think @kendallstrautman mentioned this in one of our discussion threads as well!

Reply:
She totally did and I forgot about that, apologies! It's been a while since I've done that. I think I need to use `safeAnalyticsTrack` which is in `src/lib/analytics.ts`? 🤔

Reply:
That sounds right! Pretty sure that will send to segment.

Reply:
It would be nice to get more info on the error passed to segment as well. Instead of just 'hasError', perhaps we can store some information about the error to help with debugging if we find this popping up a lot.

Reply:
Added an `error` prop to the `useAuthentication` result in https://github.com/hashicorp/dev-portal/pull/1338/commits/7a632cfeb83c768151f2280a35b7d1269d0160de; added `TODO`s for the analytics tracking in https://github.com/hashicorp/dev-portal/pull/1338/commits/fdcc27aa002f51496121ff78d7a0c0be32c6e674 (waiting to hear back from product if the values are acceptable)